src/cloud-api-adaptor/test/e2e/azure_test.go

```diff
@@ -6,142 +6,127 @@
 package e2e
 
 import (
-	"bytes"
-	"strings"
 	"testing"
 
 	_ "github.com/confidential-containers/cloud-api-adaptor/src/cloud-api-adaptor/test/provisioner/azure"
-	log "github.com/sirupsen/logrus"
 )
 
-func TestDeletePodAzure(t *testing.T) {
-	t.Parallel()
-	DoTestDeleteSimplePod(t, testEnv, assert)
-}
-
 func TestCreateSimplePodAzure(t *testing.T) {
 	t.Parallel()
 	DoTestCreateSimplePod(t, testEnv, assert)
 }
 
-func TestCreatePodWithConfigMapAzure(t *testing.T) {
-	t.Parallel()
-	DoTestCreatePodWithConfigMap(t, testEnv, assert)
-}
-
-func TestCreatePodWithSecretAzure(t *testing.T) {
-	t.Parallel()
-	DoTestCreatePodWithSecret(t, testEnv, assert)
-}
-
-func TestCreateNginxDeploymentAzure(t *testing.T) {
-	t.Parallel()
-	DoTestNginxDeployment(t, testEnv, assert)
-}
-
-func TestPodToServiceCommunicationAzure(t *testing.T) {
-	t.Parallel()
-	DoTestPodToServiceCommunication(t, testEnv, assert)
-}
-
-func TestPodsMTLSCommunicationAzure(t *testing.T) {
-	t.Parallel()
-	DoTestPodsMTLSCommunication(t, testEnv, assert)
-}
-
-func TestPodVMwithAnnotationsInstanceTypeAzure(t *testing.T) {
-	SkipTestOnCI(t)
-	t.Parallel()
-	instanceSize := "Standard_DC2as_v5"
-	DoTestPodVMwithAnnotationsInstanceType(t, testEnv, assert, instanceSize)
-}
-
-func TestPodVMwithAnnotationsInvalidInstanceTypeAzure(t *testing.T) {
-	t.Parallel()
-	// Using an instance type that's not configured in the AZURE_INSTANCE_SIZE
-	instanceSize := "Standard_D8as_v5"
-	DoTestPodVMwithAnnotationsInvalidInstanceType(t, testEnv, assert, instanceSize)
-}
-
-// Test with device annotation
-func TestPodWithCrioDeviceAnnotationAzure(t *testing.T) {
-	if !isTestOnCrio() {
-		t.Skip("Skipping test as it is not running on CRI-O")
-	}
-	t.Parallel()
-	DoTestPodWithCrioDeviceAnnotation(t, testEnv, assert)
-}
-
-// Negative test with device annotation
-func TestPodWithIncorrectDeviceAnnotationAzure(t *testing.T) {
-	if !isTestOnCrio() {
-		t.Skip("Skipping test as it is not running on CRI-O")
-	}
-	t.Parallel()
-	DoTestPodWithIncorrectCrioDeviceAnnotation(t, testEnv, assert)
-}
+// func TestCreatePodWithConfigMapAzure(t *testing.T) {
```

Comment:
Commenting out the tests will be very restrictive. For example I typically run the tests on vanilla K8s in Azure or OpenShift when testing a PR or working on a feature.

Instead of commenting out the tests, why not rename the tests in a way so that only a subset can be run as part of the github workflow using the RUN_TESTS variable.


For example we can create few groupings like
- basic
- advanced
(feel free to use any group names)

and rename the tests like
DoTestCreateSimplePodAzure

Or if you just want to run only two tests, may be run it explicitly in the workflow (much simpler)
eg
RUN_TESTS=DoTestCreateSimplePodAzure CLOUD_PROVIDER=azure make test-e2e 
RUN_TESTS=TestRemoteAttestationAzure CLOUD_PROVIDER=azure make test-e2e